docs: update best practices regex groups + windows clrf + exporting test functions (#15969)
renovatebot/renovate: docs/development/best-practices.md

- Avoid `toMatchSnapshot`, only use it for:
  - huge strings like the Renovate PR body text
  - huge complex objects where you only need to test parts

> - Avoid exporting non-exported functions purely for the purpose of testing unless it's necessary

Reviewer: ```suggestion
- Avoid exporting functions purely for the purpose of testing unless you really need to
```

> Avoid exporting non-exported functions

This is strange to me. Would my suggestion say the same thing?

Author: yes

- Omit constructors when defining Static classes
- [No `#private` fields](https://google.github.io/styleguide/tsguide.html#private-fields). instead, use TypeScript's visibility annotations
- Avoid underscore suffixes or prefixes, for example: `_prop`, use [whole words](https://google.github.io/styleguide/tsguide.html#properties-used-outside-of-class-lexical-scope) as suffix/prefix i.e. `internalProp`

### regex

Use [Named Capturing Groups](https://www.regular-expressions.info/named.html) when capturing groups (e.g `(?<groupName>CapturedGroup`)

### Windows

We recommend you set [`core.autocrlf = input`](https://git-scm.com/docs/gitattributes#_text) in your `gitConfig`, or the carriage return `\r\n` might confuse Renovate bot.
You can also set the line endings in your repository by adding `* text=auto eol=lf` to your `.gitattributes` file.

> -

Reviewer: ```suggestion

```

- Omit constructors when defining Static classes
- [No `#private` fields](https://google.github.io/styleguide/tsguide.html#private-fields). instead, use TypeScript's visibility annotations
- Avoid underscore suffixes or prefixes, for example: `_prop`, use [whole words](https://google.github.io/styleguide/tsguide.html#properties-used-outside-of-class-lexical-scope) as suffix/prefix i.e. `internalProp`

### regex

> - use [Named Capturing Groups](https://www.regular-expressions.info/named.html) when capturing groups (e.g `(?<groupName>CapturedGroup`)

Reviewer: ```suggestion
Use [Named Capturing Groups](https://www.regular-expressions.info/named.html) when capturing groups (e.g `(?<groupName>CapturedGroup`)
```

If there's one item, I don't think we should use a list. :wink: 

Are you sure about the example, it looks like you may be forgetting a `)` in the example string?

Author: i believe i wrote groups not group, but i added the word Multiple just to clear it out :D

### Windows

> - It's best if you set [core.autocrlf = input](https://git-scm.com/docs/gitattributes#_text) in your gitConfig,
>   otherwise the carriage return `\r\n` will confuse renovateBot in some cases.
>   it can also be done in your repository where you add `* text=auto eol=lf` to the `.gitattributes` file

Reviewer: ```suggestion
We recommend you set [`core.autocrlf = input`](https://git-scm.com/docs/gitattributes#_text) in your `gitConfig`, or the carriage return `\r\n` might confuse Renovate bot.
You can also set the line endings in your repository by adding `* text=auto eol=lf` to your `.gitattributes` file.
```

How about this?
